clientv3/balancer.go

```diff
@@ -137,13 +137,24 @@ func (b *simpleBalancer) updateAddrs(eps []string) {
 		addrs = append(addrs, grpc.Address{Addr: getHost(eps[i])})
 	}
 	b.addrs = addrs
+	b.mu.Unlock()
 	// updating notifyCh can trigger new connections,
 	// but balancer only expects new connections if all connections are down
-	if b.pinAddr == "" {
+	// or new addresses does not include pinAddr.
+	if !hasAddr(addrs, b.pinAddr) {
```

Comment:
this races on b.pinAddr? Probably have a separate channel for the update loop to communicate updates instead of forcing it through b.notifyCh. Something like:
```go
select {
case b.updateAddrsC <- struct{}{}:
case <-b.stopc:
}
```